src/main/java/gift/controller/KakaoAuthCodeController.java

```diff
@@ -13,14 +17,26 @@
 public class KakaoAuthCodeController {
 
     private final KakaoApiService kakaoApiService;
+    private final MemberService memberService;
+    private final JwtAuth jwtAuth;
 
-    public KakaoAuthCodeController(KakaoApiService kakaoApiService) {
+    public KakaoAuthCodeController(KakaoApiService kakaoApiService, MemberService memberService, JwtAuth jwtAuth) {
         this.kakaoApiService = kakaoApiService;
+        this.memberService = memberService;
+        this.jwtAuth = jwtAuth;
     }
 
     @GetMapping
-    public ResponseEntity<KakaoTokenResponseDto> getAccessToken(@RequestParam("code") String code) {
-        KakaoTokenResponseDto responseDto = kakaoApiService.getAccessToken(code);
+    public ResponseEntity<KakaoTokenWithJwtResponseDto> getAccessToken(@RequestParam("code") String code) {
+        KakaoTokenResponseDto kakaoTokenResponseDto = kakaoApiService.getAccessToken(code);
+        String accessToken = kakaoTokenResponseDto.accessToken();
+        String email = kakaoApiService.getUserEmail(accessToken);
```

Comment:
`kakaoApiService` 객체에게 `이메일을 알려줘` 라고 메시지를 보내볼 수 있지 않을까요? 

```suggestion
        String email = kakaoApiService.getEmail(code);
```

Reply:
`kakaoApiService` 객체에게 `이메일을 알려줘` 라고 메시지를 보내볼 수 있지 않을까요?
이건 무슨 말씀이신지 모르겠습니다.

Reply:
```
public String getEmail(String code){
    KakaoTokenResponseDto responseDto = kakaoAuth.getAccessToken(code);
    return getUserEmail(responseDto.accessToken());
}
```
이런 형태의 서비스 메서드 말씀하신거 맞나요?
이렇게 하지 않은 이유는 이후 카카오톡 메시지 전송 시 `accessToken`이 필요하고 이를 jwt 토큰에 담기 위해서 어짜피 아래의 과정이 필요하다고 생각합니다.
메서드를 추가로 만든다면 `getEmail`보다는 `accessToken`을 바로 받을 수 있는 메서드를 추가하겠습니다.
```
KakaoTokenResponseDto kakaoTokenResponseDto = kakaoApiService.getAccessToken(code);
String accessToken = kakaoTokenResponseDto.accessToken();
```